lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -320,4 +284,47 @@ sub destroy {
     $self->render(json => {result => int($ret)});
 }
 
+=over 4
+
+=item prepare_settings()
+
+Internal method to prepare settings when add or update admin table.
+Use by both B<create()> and B<update()> method.
+
+=back
+
+=cut
+
+sub prepare_settings {
+    my ($self, $table, $entry) = @_;
+    my $validation = $self->validation;
+
+    for my $par (@{$tables{$table}->{required}}) {
+        $validation->required($par);
+        if (!defined $validation->param($par)) {
+            next;
+        }
+        $entry->{$par} = trim $validation->param($par);
+    }
+
+    if ($validation->has_error) {
+        my @miss_keys = grep { $validation->has_error($_) } @{$tables{$table}->{required}};
+        return "Missing parameter: " . join(', ', @miss_keys);
```

Comment:
`return "Missing parameter: " . join(', ', @{$validation->failed});` is more efficient.

Reply:
I will modify it